tests/test_regression.py

```diff
@@ -375,3 +376,93 @@ def test_deployment(self):
         assert_equal(depl_time[np.where(agent_ids == source1_id[0])], 2)
         assert_equal(depl_time[np.where(agent_ids == source1_id[1])], 3)
 
+class TestRecycle(TestRegression):
+    """This class tests the input/recycle.xml file.
+    """
+    def __init__(self, *args, **kwargs):
+        super(TestRecycle, self).__init__(*args, **kwargs)
+
+        # this test requires separations which isn't supported by hdf5
+        # so we force sqlite:
+        base, _ = os.path.splitext(self.outf)
+        self.ext = '.sqlite'
+        self.outf = base + self.ext
+        self.inf = "../input/recycle.xml"
+        self.sql = """
+            SELECT t.time as time,SUM(c.massfrac*r.quantity) as qty FROM transactions as t
+            JOIN resources as r ON t.resourceid=r.resourceid AND r.simid=t.simid
+            JOIN agententry as send ON t.senderid=send.agentid AND send.simid=t.simid
+            JOIN agententry as recv ON t.receiverid=recv.agentid AND recv.simid=t.simid
+            JOIN compositions as c ON c.qualid=r.qualid AND c.simid=r.simid
+            WHERE send.prototype=? AND recv.prototype=? AND c.nucid=?
+            GROUP BY t.time;"""
+
+    def setUp(self):
+        super(TestRecycle, self).setUp()
+
+    def do_compare(self, fromfac, tofac, nuclide, want_invs):
+        conn = sqlite3.connect(self.outf)
+        c = conn.cursor()
+        eps = 1e-10
+        simdur = len(want_invs)
+
+        invs = [0.0]*simdur
+        for i, row in enumerate(c.execute(self.sql, (fromfac, tofac, nuclide))):
+            t = row[0]
+            invs[t] = row[1]
+
+        fname = 'want_recycle_{0}-{1}-{2}.dat'.format(fromfac, tofac, nuclide)
```

Comment:
good diagnostic tool, but I think these should be cleaned up if the test succeeds, much like our output dbs.